crates/orchestrator/src/tests/database/mod.rs

```diff
@@ -47,6 +47,187 @@ async fn test_database_create_job() -> color_eyre::Result<()> {
+    database_client.create_job(job_vec[1].clone()).await.unwrap();
+    database_client.create_job(job_vec[2].clone()).await.unwrap();
+    database_client.create_job(job_vec[3].clone()).await.unwrap();
+    database_client.create_job(job_vec[5].clone()).await.unwrap();
+    if is_successor {
+        database_client.create_job(job_vec[4].clone()).await.unwrap();
+    }
+
+    let jobs_without_successor = database_client
+        .get_jobs_without_successor(JobType::SnosRun, JobStatus::Completed, JobType::ProofCreation)
+        .await
+        .unwrap();
+
+    if is_successor {
+        assert_eq!(jobs_without_successor.len(), 0, "Expected number of jobs assertion failed.");
+    } else {
+        assert_eq!(jobs_without_successor.len(), 1, "Expected number of jobs assertion failed.");
+        assert_eq!(jobs_without_successor[0], job_vec[1], "Expected job assertion failed.");
+    }
+    Ok(())
+}
+
+/// Test for `get_last_successful_job_by_type` operation in database trait.
+/// Creates the jobs in following sequence :
+///
+/// - Creates 3 successful jobs.
+///
+/// - Should return the last successful job
+#[rstest]
+#[tokio::test]
+async fn test_database_get_last_successful_job_by_type() -> color_eyre::Result<()> {
+    TestConfigBuilder::new().build().await;
+
+    drop_database().await.unwrap();
+
+    let config = config().await;
+    let database_client = config.database();
+
+    let job_vec = [
+        build_job_item(JobType::SnosRun, JobStatus::Completed, 1),
+        build_job_item(JobType::SnosRun, JobStatus::Completed, 2),
+        build_job_item(JobType::SnosRun, JobStatus::Completed, 3),
+    ];
+
+    database_client.create_job(job_vec[0].clone()).await.unwrap();
+    database_client.create_job(job_vec[1].clone()).await.unwrap();
+    database_client.create_job(job_vec[2].clone()).await.unwrap();
+
+    let last_successful_job = database_client.get_last_successful_job_by_type(JobType::SnosRun).await.unwrap();
+
+    assert_eq!(last_successful_job.unwrap(), job_vec[2], "Expected job assertion failed");
+
+    Ok(())
+}
+
+/// Test for `get_jobs_after_internal_id_by_job_type` operation in database trait.
+/// Creates the jobs in following sequence :
+///
+/// - Creates 5 successful jobs.
+///
+/// - Should return the jobs after internal id
+#[rstest]
+#[tokio::test]
+async fn test_database_get_jobs_after_internal_id_by_job_type() -> color_eyre::Result<()> {
+    TestConfigBuilder::new().build().await;
+
+    drop_database().await.unwrap();
+
+    let config = config().await;
+    let database_client = config.database();
+
+    let job_vec = [
+        build_job_item(JobType::SnosRun, JobStatus::Completed, 1),
+        build_job_item(JobType::SnosRun, JobStatus::Completed, 2),
+        build_job_item(JobType::SnosRun, JobStatus::Completed, 3),
+        build_job_item(JobType::SnosRun, JobStatus::Completed, 4),
+        build_job_item(JobType::SnosRun, JobStatus::Completed, 5),
+    ];
```

Comment:
Nitpick: you don't really test the `by_job_type` function here since there is only one job type.

Reply:
sure I will update it in the tests.